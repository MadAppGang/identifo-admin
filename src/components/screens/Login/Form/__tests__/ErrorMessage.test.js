import LoginErrorMessage from '../ErrorMessage';

describe('<LoginErrorMessage />', () => {
  test('renders as expected', () => {
    const props = {
      onClick: jest.fn(),
      message: 'Hello, world!',
    };

    expect(shallow(<LoginErrorMessage {...props} />)).toMatchSnapshot();
  });

  test('calls onClick prop on button click', () => {
    const onClick = jest.fn();

    shallow(<LoginErrorMessage message="msg" onClick={onClick} />)
      .find('button')
      .simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
