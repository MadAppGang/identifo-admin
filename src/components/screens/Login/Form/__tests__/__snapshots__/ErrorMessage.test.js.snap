// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginErrorMessage /> renders as expected 1`] = `
<div
  className="iap-login-form__error-msg"
>
  <button
    onClick={[MockFunction]}
    tabIndex={-1}
    type="button"
  >
    Hello, world!
  </button>
</div>
`;
